tests/components/devolo_home_network/test_binary_sensor.py

```diff
@@ -24,19 +22,23 @@
 
 
 @pytest.mark.usefixtures("mock_device")
-async def test_binary_sensor_setup(hass: HomeAssistant) -> None:
+async def test_binary_sensor_setup(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+) -> None:
     """Test default setup of the binary sensor component."""
     entry = configure_integration(hass)
     device_name = entry.title.replace(" ", "_").lower()
     await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
+    assert entry.state is ConfigEntryState.LOADED
 
-    assert (
-        hass.states.get(f"{BINARY_SENSOR_DOMAIN}.{device_name}_{CONNECTED_TO_ROUTER}")
-        is None
-    )
+    assert entity_registry.async_get(
+        f"{PLATFORM}.{device_name}_connected_to_router"
+    ).disabled
 
     await hass.config_entries.async_unload(entry.entry_id)
+    assert entry.state is ConfigEntryState.NOT_LOADED
```

Comment:
Remind me why did we unload these again?

Reply:
The unload was there, but in basically every test and I thought your comment

` don't believe we have to

    await hass.config_entries.async_unload(entry.entry_id)

every test`

meant to keep it at least somewhere. Should I remove it completely?

Reply:
That test was indeed already there :+1: